envoyauth/evaluation.go

```diff
@@ -0,0 +1,125 @@
+package envoyauth
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/bundle"
+	"github.com/open-policy-agent/opa/metrics"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/storage"
+	iCache "github.com/open-policy-agent/opa/topdown/cache"
+	"github.com/sirupsen/logrus"
+)
+
+//EvalContext - This is an SPI that has to be provided if the envoy external authorization
+// is used from outside the plugin, i.e. as a Go module
+type EvalContext interface {
+	ParsedQuery() ast.Body
+	Store() storage.Store
+	Compiler() *ast.Compiler
+	Runtime() *ast.Term
+	PreparedQueryDoOnce() *sync.Once
+	InterQueryBuiltinCache() iCache.InterQueryCache
+	PreparedQuery() *rego.PreparedEvalQuery
+	SetPreparedQuery(*rego.PreparedEvalQuery)
+}
+
+//Eval - Evaluates an input against a provided EvalContext and yields result
+func Eval(ctx context.Context, evalContext EvalContext, input ast.Value, result *EvalResult, opts ...func(*rego.Rego)) error {
+
+	err := storage.Txn(ctx, evalContext.Store(), storage.TransactionParams{}, func(txn storage.Transaction) error {
+
+		var err error
+
+		err = getRevision(ctx, evalContext.Store(), txn, result)
+		if err != nil {
+			return err
+		}
+
+		result.TxnID = txn.ID()
+
+		logrus.WithFields(logrus.Fields{
+			"input": input,
```

Comment:
💭 `input` could be pretty huge. But it's debug, so, fine.

Reply:
Same here: https://github.com/open-policy-agent/opa-envoy-plugin/blob/42b3ce43573c94b02259a2b020bcdae3cb24c108/internal/internal.go#L432-L433